src/frontend/sharedTypes.ts

```diff
@@ -0,0 +1,41 @@
+// TS port of /src/frontend/types.js
```

Comment:
open to renaming this or doing more of a directory-style type of structure with this. used this name to differentiate it from `types.js`, which is still needed as long as Flow is used in the repo